src/main/java/com/streamr/client/authentication/Session.java

```diff
@@ -10,15 +10,11 @@
     private final AuthenticationMethod authenticationMethod;
     private String sessionToken = null;
 
-    /**
-     * For unauthenticated sessions
-     */
-    public Session() {
-        authenticationMethod = null;
-    }
-
-    public Session(AuthenticationMethod authenticationMethod) {
+    public Session(String restApiUrl, AuthenticationMethod authenticationMethod) {
         this.authenticationMethod = authenticationMethod;
+        if (this.authenticationMethod != null) {
+            this.authenticationMethod.setRestApiUrl(restApiUrl);
```

Comment:
This bit smells hacky to me. I feel that the `setRestApiUrl` method should be removed because passing this value is required to use the class, see earlier comment.